src/rgw/rgw_aws.cc

```diff
@@ -155,7 +156,12 @@ namespace rgw::aws {
       Aws::Utils::ARN arn((Aws::String(resourceArn)));
       invokeRequest.SetInvocationType(Aws::Lambda::Model::InvocationType::RequestResponse);
       invokeRequest.SetLogType(Aws::Lambda::Model::LogType::Tail);
-      invokeRequest.SetFunctionName(arn.GetResource());
+      boost::optional<ARNResource> resourceParsed = ARNResource::parse(std::string(arn.GetResource()));
```

Comment:
can we do all of that (including line 156) in the external "publish" function?
if the ARN is malformed we should fail there.
even better, can we do that in the "connect" function?
I think that the ARN is constant per endpoint, and then we will do the parsing once per connection instead of once per record?